testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -0,0 +1,517 @@
+    ) -> Result<ThresholdSignature, NakamotoNodeError> {
+        let sign_id = Self::get_sign_id(burn_tip.block_height, burnchain);
+        let sign_iter_id = block_attempt;
+        let reward_cycle_id = burnchain
+            .block_height_to_reward_cycle(burn_tip.block_height)
+            .expect("FATAL: tried to initialize coordinator before first burn block height");
+        self.coordinator.current_sign_id = sign_id;
+        self.coordinator.current_sign_iter_id = sign_iter_id;
+
+        let block_bytes = block.serialize_to_vec();
+        let nonce_req_msg = self
+            .coordinator
+            .start_signing_round(&block_bytes, false, None)
+            .map_err(|e| {
+                NakamotoNodeError::SigningCoordinatorFailure(format!(
+                    "Failed to start signing round in FIRE coordinator: {e:?}"
+                ))
+            })?;
+        Self::send_signers_message(
+            &self.message_key,
+            sortdb,
+            burn_tip,
+            stackerdbs,
+            nonce_req_msg.into(),
+            self.is_mainnet,
+            &self.miners_db_config,
+            event_dispatcher,
+        )
+        .map_err(NakamotoNodeError::SigningCoordinatorFailure)?;
+
+        let Some(ref mut receiver) = self.receiver else {
+            return Err(NakamotoNodeError::SigningCoordinatorFailure(
+                "Failed to obtain the StackerDB event receiver".into(),
+            ));
+        };
+
+        let start_ts = Instant::now();
+        while start_ts.elapsed() <= self.signing_round_timeout {
+            let event = match receiver.recv_timeout(Duration::from_millis(50)) {
+                Ok(event) => event,
+                Err(std::sync::mpsc::RecvTimeoutError::Timeout) => {
+                    continue;
+                }
+                Err(std::sync::mpsc::RecvTimeoutError::Disconnected) => {
+                    return Err(NakamotoNodeError::SigningCoordinatorFailure(
+                        "StackerDB event receiver disconnected".into(),
+                    ))
+                }
+            };
+
+            let is_signer_event = event.contract_id.name.starts_with(SIGNERS_NAME)
+                && event.contract_id.issuer.1 == [0; 20];
+            if !is_signer_event {
+                debug!("Ignoring StackerDB event for non-signer contract"; "contract" => %event.contract_id);
+                continue;
+            }
+            let Ok(signer_event) = SignerEvent::try_from(event).map_err(|e| {
+                warn!("Failure parsing StackerDB event into signer event. Ignoring message."; "err" => ?e);
+            }) else {
+                continue;
+            };
+            let SignerEvent::SignerMessages(signer_set, messages) = signer_event else {
+                debug!("Received signer event other than a signer message. Ignoring.");
+                continue;
+            };
+            if signer_set != u32::try_from(reward_cycle_id % 2).unwrap() {
```

Comment:
In a case like this, I feel like it's cleaner to use an `as u32` instead of a `try_from` + `unwrap`.

Reply:
I kind of agree, but would like to see some lints activated before switching from defaulting to `try_from().unwrap()`:

```
#![warn(
    clippy::cast_possible_wrap, // unsigned -> signed
    clippy::cast_sign_loss, // signed -> unsigned
    clippy::cast_lossless,
    clippy::cast_possible_truncation,
)]
```

That way, at least CI would tell me if my reasoning about modulus arithmetic was wrong.